src/test/java/io/javarig/generation/javapredefinedtypes/collection/SingleGenericTypeCollectionGenerationTest.java

```diff
@@ -9,93 +9,105 @@
+    public void shouldReturnTheAppropriateCollectionWithSizeBetween(Class<?> collectionClass) {
         //given
         int minSize = 20;
         int maxSize = 40;
         Class<?> type = String.class;
         //when
-        Object generated = randomInstanceGenerator.generate(List.class, minSize, maxSize, type);
+        Object generated = randomInstanceGenerator.generate(collectionClass, minSize, maxSize, type);
         //then
         log.info("shouldReturnListWithSizeBetween : {}", generated);
         assertThat(generated).isNotNull();
-        assertThat(generated).isInstanceOf(List.class);
-        assertThat(generated).asInstanceOf(LIST).hasSizeBetween(minSize, maxSize);
+        assertThat(generated).isInstanceOf(collectionClass);
+        assertThat(generated).asInstanceOf(COLLECTION).hasSizeBetween(minSize, maxSize);
 
         //asserting type
-        assertThat(generated).asInstanceOf(LIST).first().isInstanceOf(type);
+        assertThat(generated).asInstanceOf(COLLECTION).first().isInstanceOf(type);
     }
-    @Test
-    public void shouldThrowInvalidGenericParamsNumberExceptionWhenTryingToGenerateAListWithAZeroGenericParams() {
+    @ParameterizedTest
+    @MethodSource("getImplementedCollectionClasses")
+    public void shouldThrowInvalidGenericParamsNumberExceptionWhenTryingToGenerateACollectionWithAZeroGenericParams(Class<?> collectionClass) {
         int required = 1;
-        assertThatThrownBy(() -> randomInstanceGenerator.generate(List.class))
+        assertThatThrownBy(() -> randomInstanceGenerator.generate(collectionClass))
                 .isInstanceOf(InvalidGenericParametersNumberException.class)
                 .hasMessage("invalid number of generic parameters, required %d and 0 was found".formatted(required));
     }
 
-    @Test
-    public void shouldThrowInvalidGenericParamsNumberExceptionWhenTryingToGenerateAListWithAMoreThanOneGenericParams() {
+    @ParameterizedTest
+    @MethodSource("getImplementedCollectionClasses")
+    public void shouldThrowInvalidGenericParamsNumberExceptionWhenTryingToGenerateACollectionWithAMoreThanOneGenericParams(Class<?> collectionClass) {
         int required = 1;
         Type[] genericParams = new Type[]{String.class, Integer.class};
-        assertThatThrownBy(() -> randomInstanceGenerator.generate(List.class, genericParams))
+        assertThatThrownBy(() -> randomInstanceGenerator.generate(collectionClass, genericParams))
                 .isInstanceOf(InvalidGenericParametersNumberException.class)
-                .hasMessage("invalid number of generic parameters, required %d and %d was found".formatted(required, genericParams.length));
+                .hasMessage("invalid number of generic parameters, required %d and %d was found".formatted(required, genericParams.length)); // FIXME is this how should we check for message ? it would be better if we checked a type rather than a string that could be changed in the future without necessarily breaking the code
```

Comment:
We check the type before checking the message, and this message check could be removed without breaking anything, however, I think we better keep it to have a stronger test.
If the message is changed, then the test will immediately break, and the change won't be merged if the message was not changed in the test too, so no code break will happen